### PR TITLE
Add ability to configure explain command options

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

**Review by blink1073 — APPROVED** [review]
LGTM!

File: test/test_collection.py [hunk 1]
```diff
@@ -116,13 +132,13 @@ def test_delete_many(self):
...
         res = self.explain.watch()
-        self.assertIn("queryPlanner", res["stages"][0]["$cursor"])
+        self.assertTrue(self._recursiveIn("queryPlanner", res))
         self.collection.watch(pipeline=[{"$project": {"tags": 1}}],
                               batch_size=10, full_document="updateLookup")
         last_logger_payload = self.logger.cmd_payload
         res = self.explain.watch(pipeline=[{"$project": {"tags": 1}}],
                                  batch_size=10, full_document="updateLookup")
-        self.assertIn("queryPlanner", res["stages"][0]["$cursor"])
+        self.assertTrue(self._recursiveIn("queryPlanner", res))
```
Comment (ShaneHarvey) [inline test/test_collection.py:141]: I actually prefer the previous behavior here because `res["stages"][0]["$cursor"]` helps us understand the structure of the explain response. `_recursiveIn` makes the command response more of a black box. Can you revert these changes?

File: test/test_collection.py [hunk 2]
```diff
@@ -217,6 +231,29 @@ def test_imports(self):
...
+        self.explain.update_settings(verbosity="executionStats")
+        res = self.explain.find({})
+        self.assertTrue(self._recursiveIn("executionStats", res))
+        self.assertFalse(self._recursiveIn("allPlansExecution", res))
+        self.explain.update_settings(verbosity="allPlansExecution")
+        res = self.explain.find({})
+        self.assertTrue(self._recursiveIn("executionStats", res))
+        self.assertTrue(self._recursiveIn("allPlansExecution", res))
+
+    def test_comment(self):
```
Comment (ShaneHarvey) [inline test/test_collection.py:247]: Instead of checking the explain result, this test should probably assert that we sent the comment field with the explain comment using a CommandListener.

File: pymongoexplain/explainable_collection.py [hunk 3]
```diff
@@ -26,17 +26,27 @@
...
         self.collection = collection
         self.last_cmd_payload = None
+        self.verbosity = verbosity
+        self.comment = comment
 
     def _explain_command(self, command):
         command_son = command.get_SON()
         explain_command = SON([("explain", command_son)])
-        explain_command["verbosity"] = "queryPlanner"
+        explain_command["verbosity"] = self.verbosity or "queryPlanner"
```
Comment (ShaneHarvey) [inline pymongoexplain/explainable_collection.py:38]: I suggest setting the default verbosity in `__init__`.

File: pymongoexplain/explainable_collection.py [hunk 4]
```diff
@@ -26,17 +26,27 @@
...
+        if self.comment:
+            explain_command["comment"] = self.comment
         self.last_cmd_payload = command_son
         return self.collection.database.command(explain_command)
 
+    def update_settings(self, verbosity=None, comment=None):
```
Comment (ShaneHarvey) [inline pymongoexplain/explainable_collection.py:44]: In PyMongo the Collection object is immutable (the internal settings are configured once at creation and never changed). ExplainableCollection should probably be immutable too to match pymongo.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Are there any docs or changelog that should go along with this change?

File: test/test_collection.py (hunk 2 above)
Comment (juliusgeo) [reply test/test_collection.py:234]: Done.

File: test/test_collection.py (hunk 1 above)
Comment (juliusgeo) [reply test/test_collection.py]: Done.

File: pymongoexplain/explainable_collection.py (hunk 4 above)
Comment (juliusgeo) [reply pymongoexplain/explainable_collection.py]: Done.

File: pymongoexplain/explainable_collection.py (hunk 3 above)
Comment (juliusgeo) [reply pymongoexplain/explainable_collection.py]: Done.

File: test/test_collection.py [hunk 5]
```diff
@@ -114,14 +114,15 @@ def test_delete_many(self):
         last_cmd_payload = self.explain.last_cmd_payload
         self._compare_command_dicts(last_cmd_payload, last_logger_payload)
 
+    @unittest.skip("Travis does not have replica sets set up yet")
```
Comment (ShaneHarvey) [inline test/test_collection.py:117]: This is no longer true.

File: README.rst [hunk 6]
```diff
@@ -93,6 +93,15 @@ To run explain on a command, first instantiate an ``ExplainableCollection`` from
     collection = client.db.products
     explain = ExplainableCollection(collection)
 
+If you wish to configure the options for the explain command itself, pass
+them to the ``ExplainableCollection`` constructor like so::
+
+    explain = ExplainableCollection(collection, verbosity="queryPlanner",
+                                    comment="I'm a comment")
+For more information see the documentation for the explain_ command.
```
Comment (ShaneHarvey) [inline README.rst:101]: I believe there needs to be a blank line after the code block.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: test/test_collection.py (hunk 5 above)
Comment (juliusgeo) [reply test/test_collection.py:117]: Done.

File: test/test_collection.py [hunk 7]
```diff
@@ -118,10 +119,10 @@ def test_watch(self):
         res = self.explain.watch()
         self.assertIn("queryPlanner", res["stages"][0]["$cursor"])
         self.collection.watch(pipeline=[{"$project": {"tags": 1}}],
-                              batch_size=10, full_document="updateLookup")
+                               batch_size=10, full_document="updateLookup")
```
Comment (ShaneHarvey) [inline test/test_collection.py:122]: This change seems off batch_size should align with pipeline.

File: test/test_collection.py [hunk 8]
```diff
@@ -118,10 +119,10 @@ def test_watch(self):
...
         last_logger_payload = self.logger.cmd_payload
         res = self.explain.watch(pipeline=[{"$project": {"tags": 1}}],
-                                 batch_size=10, full_document="updateLookup")
+                            batch_size=10, full_document="updateLookup")
```
Comment (ShaneHarvey) [inline test/test_collection.py:125]: Same here.

File: test/test_collection.py [hunk 9]
```diff
@@ -217,6 +218,27 @@ def test_imports(self):
...
+        res = self.explain.find({})
+        self.assertIn("executionStats", res)
+        self.assertIn("allPlansExecution", res["executionStats"])
+
+    def test_comment(self):
+        self.explain.find({})
+        self.assertNotIn("comment", self.logger.cmd_payload)
+        self.explain = ExplainCollection(self.collection, comment="hey, "
+                                                                  "I'm a "
+                                                                  "comment")
```
Comment (ShaneHarvey) [inline test/test_collection.py:239]: Please fix the string format here. 3 short strings on new lines makes this harder to read. How about:
```
self.explain = ExplainCollection(self.collection, comment="comment")
```

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: test/test_collection.py (hunk 9 above)
Comment (juliusgeo) [reply test/test_collection.py:239]: Done.

File: test/test_collection.py (hunk 7 above)
Comment (juliusgeo) [reply test/test_collection.py:122]: Done.

File: test/test_collection.py (hunk 8 above)
Comment (juliusgeo) [reply test/test_collection.py:125]: Done.

**Review by ShaneHarvey — APPROVED** [review]
